### DISCUSSION TITLE
[0.76] iOS `minimum OS` version bump to `15.1` Announcement

### DISCUSSION THREAD

**Discussion by realsoelynn** [react-native-community/discussions-and-proposals#812] [Announcements]
**TL;DR**

From React Native `0.76` the minimum iOS version will be `15.1`.

---

We're sharing this announcement to inform our community that we're about to bump the minimum iOS version for React Native.  React Native `0.76` will be the first version with a minimum iOS version of `15.1`. 

We're sharing this announcement now to give enough time to update your app and communicate it to your users..

## Why are we jumping two version bumps instead of going to `iOS 14`?
Both iOS `14` and `15` support the same minimum iPhone which is `iPhone 6s`.

## Why we're doing this change
There are two major factors we consider for bumping the minSdk amongst other things;
- sufficient test coverage for old OSes
- developer resources required to maintain the legacy code

By dropping support for older versions of iOS, we're able to offer you a better React Native experience and simplify our codebase.

## What does this mean for me?
We believe this change won't affect most developers and is part of the natural mobile update cycles.

In the rare circumstance where you're targeting older devices, this change means that users of your applications on `iPhone 6` and below won't be able to receive updates of your apps if you update your project beyond React Native `0.76`. In those cases the recommendation is to use React Native `0.75` as long as possible, or work out a plan to update the minimum iOS version of your app.

If you have not migrated to the new architecture but plan to do so, then we would suggest testing it out ASAP as you will be stuck with `0.75` for a while. You should report any critical bugs early so it can be fixed while the [support window](https://github.com/reactwg/react-native-releases/blob/main/docs/support.md) is still open.

**Comment by fabiendem** [comment]
About:
> If you have not migrated to the new architecture but plan to do so, then we would suggest testing it out ASAP as you will be stuck with 0.75 for a while.

Do you have more info about this or a link to get more details?
It's the first time I read the signal that one version of React Native (0.76+) would not be retro-compatible with the old architecture (at least that's how we can interpret this).

Thanks!

**Reply by realsoelynn** [reply]
I am referring to the App that still want to support min iOS 13.4 AND want to enable New Architecture.

0.76+ will still be retro-compatible with old architecture. Nothing is changing for that part.

Thank you so much for asking this clarifying question. I will update the last paragraph so that it does NOT cause confusion.

**Reply by fabiendem** [reply]
Thanks a lot for the prompt and clear answer ❤️

**Reply by just-myself** [reply]
I want to confirm, can I still be compatible with 13.4 after upgrading to the 0.76 iOS version?
I have a large number of low version users, is it still compatible with 13.4?

**Reply by cortinico** [reply]
> can I still be compatible with 13.4 after upgrading to the 0.76 iOS version?

Nope. Version 0.76 bumps the min iOS version to 15.1

**Comment by KingAmo** [comment]
> Both iOS 14 and 15 support the same minimum iPhone which is iPhone 6s.

This is not sufficient reason to skip `iOS 14`，we have a significant amount of user on `iOS 14`

**Reply by fujufutong** [reply]
agree

**Reply by realsoelynn** [reply]
Hi there, 

We understand and empathize with the fact mobile has usually a long tail of users that sticks with older versions of the system and that is a problem when these kind of updates happens.

The last version of iOS 14 is `14.8.1` and it was released on `October, 2021`. Apple has stopped `iOS 14` for security update for almost `3 years` now. It's important to stress that same minimum iPhone is supported between iOS 14 and 15.
All the users on `iOS 14` can update to `iOS 15.1` seamlessly.

Another reason for the bump is that React Native is also used internally and we had to follow the internal policies. All internal apps needs to be migrated to `iOS 15.1` so we need to operate accordingly.

As a mitigation, we can suggest a couple of approaches:
1. You can stay on `0.75` as long as the version is in the support window, to give users the time to update their system. You don't have to adopt `0.76` right away when we will release it.
2. You can use these months to prompt your users to update their system as in a future version `iOS 14` would not be supported anymore. Notice that if you publish a version that is not compatible with the system of a user, the user will still continue to use the app with an older version.

**Reply by KingAmo** [reply]
> Notice that if you publish a version that is not compatible with the system of a user, the user will still continue to use the app with an older version.

This is true only when the Apple ID (Apple Account) has downloaded(Purchased) the app before, otherwise your Apple ID will never be able to download your app in App Store

**Reply by fujufutong** [reply]
why 15.1 not 15.0?

**Reply by douglasjunior** [reply]
Unfortunately, the maintainers do not live in third world countries, so they cannot see the value in maintaining a lower requirement.

This has always been the case, and it is not only in the React Native community.

Here in Brazil, a new iPhone costs 3 months of salary, while in the US the same device costs a few hours.

**Reply by radko93** [reply]
@douglasjunior all devices that supports iOS 14 can run iOS 15

**Reply by douglasjunior** [reply]
My comment is not about iOS 14 or 15.

**Comment by fujufutong** [comment]
disagree, iOS14 is popular and iOS 14 is smoother than iOS15

**Reply by cipolleschi** [reply]
Hi @fujufutong, see the reply [here](https://github.com/react-native-community/discussions-and-proposals/discussions/812#discussioncomment-10361821) for a more detailed explanation.

Also, you mention that iOS 14 is popular, but I looked online and found opposite evidences ([here](https://telemetrydeck.com/blog/ios-market-share-04-24/) and [here](https://www.statista.com/statistics/1118925/mobile-apple-ios-version-share-worldwide/)). I'm sure you have access to some dataset I can't see. Do you mind sharing the dataset or why you think that 14 is popular?

**Reply by fujufutong** [reply]
why 15.1 not 15.0?

**Comment by fujufutong** [comment]
why 15.1 not 15.0?

**Reply by cipolleschi** [reply]
Hi @fujufutong, thanks for the great question.
Internally, we never adopt X.0 versions because we have seen that they are slightly unstable. The X.1 version usually contains all the stability fixes to make the version more reliable and therefore better suited to be a good min version.
